contracts/lib/ExtensionAgglayerBridgeL2.sol

```diff
@@ -0,0 +1,587 @@
+        globalExitRootManager = _globalExitRootManager;
+        polygonRollupManager = _polygonRollupManager;
+        bridgeManager = _bridgeManager;
+        emergencyBridgePauser = _emergencyBridgePauser;
+        emit AcceptEmergencyBridgePauserRole(address(0), emergencyBridgePauser);
+        emergencyBridgeUnpauser = _emergencyBridgeUnpauser;
+        emit AcceptEmergencyBridgeUnpauserRole(
+            address(0),
+            emergencyBridgeUnpauser
+        );
+
+        // Set proxied tokens manager
+        require(
+            _proxiedTokensManager != address(this),
+            BridgeAddressNotAllowed()
+        );
+
+        // It's not allowed proxiedTokensManager to be zero address. If disabling token upgradability is required, add a not owned account like 0xffff...fffff
+        require(_proxiedTokensManager != address(0), InvalidZeroAddress());
+
+        proxiedTokensManager = _proxiedTokensManager;
+
+        emit AcceptProxiedTokensManagerRole(address(0), proxiedTokensManager);
+
+        // Set gas token
+        if (_gasTokenAddress == address(0)) {
+            // Gas token will be ether
+            if (_gasTokenNetwork != 0) {
+                revert GasTokenNetworkMustBeZeroOnEther();
+            }
+            // Health check for sovereign WETH address
+            if (
+                _sovereignWETHAddress != address(0) ||
+                _sovereignWETHAddressIsNotMintable
+            ) {
+                revert InvalidSovereignWETHAddressParams();
+            }
+            // WETHToken, gasTokenAddress and gasTokenNetwork will be 0
+            // gasTokenMetadata will be empty
+        } else {
+            // Gas token will be an erc20
+            gasTokenAddress = _gasTokenAddress;
+            gasTokenNetwork = _gasTokenNetwork;
+            gasTokenMetadata = _gasTokenMetadata;
+
+            // Set sovereign weth token or create new if not provided
+            if (_sovereignWETHAddress == address(0)) {
+                // Health check for sovereign WETH address is mintable
+                if (_sovereignWETHAddressIsNotMintable == true) {
+                    revert InvalidSovereignWETHAddressParams();
+                }
+                // Create a wrapped token for WETH, with salt == 0
+                WETHToken = _deployWrappedToken(
+                    0, // salt
+                    abi.encode("Wrapped Ether", "WETH", 18)
+                );
+            } else {
+                WETHToken = ITokenWrappedBridgeUpgradeable(
+                    _sovereignWETHAddress
+                );
+                wrappedAddressIsNotMintable[
+                    _sovereignWETHAddress
+                ] = _sovereignWETHAddressIsNotMintable;
+            }
+        }
+
+        // Initialize OZ contracts
+        __ReentrancyGuard_init();
+    }
+
+    ///////////////////////////////////////////////////////////////
+    // OVERRIDE ALL INHERITED FUNCTIONS TO REVERT - BYTECODE OPTIMIZATION
```

Comment:
function initialize(
        uint32, // _networkID
        address, //_gasTokenAddress
        uint32, //_gasTokenNetwork
        IBaseLegacyAgglayerGER, //_globalExitRootManager
        address, //_polygonRollupManager
        bytes memory //_gasTokenMetadata
    )
Although is already being overwritten in AgglayerBridgeL2 I would consider doing it in this Extension contract in the same way